### DISCUSSION TITLE
[Question][Hw4] Evaluation of coverage

### DISCUSSION THREAD

**Discussion by AnHaechan** [prosyslab-classroom/cs348-information-security#578] [Q&A]
Will evaluation of coverage be done in the `json_parser.c` with hidden bugs?
I ask this question because for some control flow of instructions, if injected bug occurs in the front of that control flow,
the rest instructions might not be covered.
Therefore, our coverage for `json_parser.c` with hidden bugs can be a lot lower than the coverage from the public one.
Actually I've injected bugs in random places in the code and resulted in much lower coverage.
Or, are the injected bugs somehow placed at the position that does not affect the coverage that much?

**Comment by RiceBiscuits** [comment]
To avoid the problem you mentioned, TA tried to carefully choose bugs, but if your seed become complex, it can be. Thus, evaluated coverage can be decreased comparing to your expectation.

**Comment by AnHaechan** [comment]
Thanks for the answer!
